File: backend/model/lead.go
package model

import "time"

type Lead struct {
    ID           int        `gorm:"primaryKey;autoIncrement" json:"id"`
    LeadUUID     string     `gorm:"column:lead_uuid" json:"lead_uuid"`
    Name         string     `gorm:"column:name" json:"name"`
    MobileNumber string     `gorm:"column:mobile_number" json:"mobile_number"`
    Email        string     `gorm:"column:email" json:"email"`
    EnquiryText  string     `gorm:"column:enquiry_text" json:"enquiry_text"`
    UniqueLeadID string     `gorm:"column:unique_lead_id" json:"unique_lead_id"`
    LeadStatus   string     `gorm:"column:lead_status" json:"lead_status"`
    Services     string     `gorm:"column:services" json:"services"`
    CreatedAt    time.Time  `gorm:"column:created_at" json:"created_at"`
    UpdatedAt    time.Time  `gorm:"column:updated_at" json:"updated_at"`
    DeletedAt    *time.Time `gorm:"column:deleted_at" json:"deleted_at"`
}

func (Lead) TableName() string { return "leads" }

File: frontend/src/utils/dto/response/lead.ts
export type LeadListItem = {
    name: string;
    mobile: string;
    email: string;
    enquiry: string;
    status: string;
    created_at: string;
    id: number;
    lead_uuid: string;
};

export type LeadDetails = {
    id: number;
    lead_uuid: string;
    name: string;
    mobile_number: string;
    unique_lead_id: string;
    lead_status: string;
    services: string;
    created_at: string;
    updated_at: string;
};

export type LeadComment = {
    id: number;
    lead_id: number;
    reamrk: string;
    type: string;
    user_name: string;
    assigned_to: string;
    created_at: string;
};

export type LeadActivity = {
    id: number;
    lead_id: number;
    user_id: number;
    user_name: string;
    remarks: string;
    type: string;
    assigned_to: string;
    assigned_from: string;
    created_at: string;
    updated_at: string;
    deleted_at: string | null;
};

export type LeadFollowUp = {
    id: number;
    uuid: string;
    status: string;
    remark: string;
    follow_up_date: string;
    follow_up_time: string;
    next_follow_up_date: string;
    next_follow_up_time: string;
    lead_status: string;
    customer_name: string;
    lead_uuid: string;
    lead_mobile_number: string;
    lead_unique_id: string;
    created_at: string;
    updated_at: string;
    updated_by: string;
};
